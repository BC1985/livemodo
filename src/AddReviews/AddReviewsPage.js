import React, { useState } from "react";
import {apiService} from "../services/auth-api-service";
import { TextField } from "formik-material-ui";
import { Formik, Form, Field } from "formik";
import { makeStyles } from "@material-ui/core/styles";
import { KeyboardDatePicker } from "@material-ui/pickers";
import { Button, Container, CircularProgress } from "@material-ui/core";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";

function AddreviewsPage() {
  const [hasErrors, setHasErrors] = useState({});

  const useStyles = makeStyles(theme => ({
    paper: {
      marginTop: theme.spacing(8),
      marginBottom: 100,
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
    },
    form: {
      width: "100%",
      marginTop: theme.spacing(8),
      marginBottom: theme.spacing(8),
    },
    submit: {
      margin: theme.spacing(3, 0, 2),
    },
  }));
  const classes = useStyles();
  // show any error message if field is populated
  const errorMessage = Object.values(hasErrors).filter(err => err.length !== 0);

  return (
    <Container component="main" maxWidth="xs">
      <div className={classes.paper}>
        <Typography component="h1" variant="h5" style={{ margin: "10% 0 15%" }}>
          Post your review
        </Typography>

        <Formik
          initialValues={{
            tagline: "",
            bandName: "",
            venue: "",
            content: "",
            showDate:"",
            rating: ""
          }}
          enableReinitialize
          validate={values => {
            const {
              email,
              tagline,
              bandName,
              content,
              rating,
              showDate
            } = values;
            const errors = {};
            // // validate email
            // if (!email) {
            //   errors.email = "Required";
            // } 
            // // // // validate username
            // if (!content) {
            //   errors.username = "Please type your review";
            // } else if (content.length < 2) {
            //   errors.username = "must be at least two characters.";
            // }
            // return errors;
          }}
          onSubmit={async (values, actions) => {
            actions.setSubmitting(true);
            const submitReview = await apiService.postReview({
              tagline: values.tagline.trim(),
              bandName: values.bandName.trim(),
              venue: values.venue.trim(),
              content: values.content.trim(),
              showDate: values.showDate.trim(),
              rating: values.rating.trim(),
            });
            // if (submitReview.errors) {
            //   setHasErrors(apiCall.errors);
            //   actions.setSubmitting(false);
            // } else {
            //   actions.setSubmitting(true);
            //   history.push("/");
            // }
          }}
        >
          {({ submitForm, isSubmitting, isValid, dirty, setFieldValue }) => (
            <Form>
              <div>
                {isSubmitting && (
                  <CircularProgress
                    size={60}
                    variant="indeterminate"
                    id="spinner"
                  />
                )}
              </div>
              <Grid container spacing={2}>
                <Grid item>
                  <Field
                    component={TextField}
                    name="tagline"
                    variant="outlined"
                    placeholder="optional"
                    fullWidth
                    id="tagline"
                    label="tagline"
                    autoFocus
                  />
                </Grid>
                <Grid item xs={12} sm={6}>
                  <Field
                    component={TextField}
                    variant="outlined"
                    required
                    fullWidth
                    id="bandName"
                    label="Name of artist or band"
                    name="bandName"
                  />
                </Grid>
                <Grid item>
                  <Field
                    component={TextField}
                    variant="outlined"
                    type="content"
                    required
                    fullWidth
                    id="content"
                    label="What did you think?"
                    name="content"
                  />
                </Grid>
                <Grid item>
                  <Field
                    component={KeyboardDatePicker}
                    label="Show date"
                    name="showDate"
                    disableFuture={true}
                    onChange={val => {
                      console.log(val);
                      setFieldValue("dob", val);
                    }}
                  />
                </Grid>
              </Grid>
              <Button
                onClick={submitForm}
                type="submit"
                fullWidth
                disabled={!isValid || isSubmitting }
                variant="contained"
                color="default"
                className={classes.submit}
              >
                Sign Up
              </Button>
            </Form>
          )}
        </Formik>
        <p style={{ color: "red" }}>{errorMessage}</p>
      </div>
    </Container>
  );
}

export default AddreviewsPage;
