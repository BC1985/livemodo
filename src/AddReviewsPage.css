.add-review-container {
  min-height: 100%;
  margin-bottom: -50px;
}

.push {
  height: 50px;
}

#add-review-header {
  padding: 20px;
}
