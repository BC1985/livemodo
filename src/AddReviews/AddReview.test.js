import React from "react";
import ReactDOM from "react-dom";
import AddReviewsPage from "../AddReviews/AddReviewsPage";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <AddReviewsPage />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
