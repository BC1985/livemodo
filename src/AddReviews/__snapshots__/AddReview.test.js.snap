// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <div
    id="add-review-header"
  >
    <h2>
      Add review
    </h2>
  </div>,
  <div
    className="add-review-container"
  >
    <div />
    <form
      id="add-review-form"
      onSubmit={[Function]}
    >
      <label>
        Add tagline 
        <span
          style={
            Object {
              "color": "smokewhite",
              "fontStyle": "italic",
            }
          }
        >
          (optional)
        </span>
      </label>
      <input
        name="tagline"
        onChange={[Function]}
        placeholder="One line to describe your experience"
        type="text"
        value=""
      />
      <label>
        Name of performer/ band*
      </label>
      <input
        className={null}
        name="band_name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="e.g The Hungry Caterpillars"
        required={true}
        value=""
      />
      <label>
        Name of venue*
      </label>
      <input
        className={null}
        name="venue"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="e.g Yolanda's Prophylactic Emporium"
        required={true}
        value=""
      />
      <label>
        Date of performace*
      </label>
      <div
        id="calendar"
      >
        <div
          className="react-calendar"
          onBlur={null}
          onMouseOut={null}
        >
          <div
            className="react-calendar__navigation"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <button
              aria-label=""
              className="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
              onClick={[Function]}
              type="button"
            >
              «
            </button>
            <button
              aria-label=""
              className="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
              onClick={[Function]}
              type="button"
            >
              ‹
            </button>
            <button
              aria-label=""
              className="react-calendar__navigation__label"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
              type="button"
            >
              October 2019
            </button>
            <button
              aria-label=""
              className="react-calendar__navigation__arrow react-calendar__navigation__next-button"
              onClick={[Function]}
              type="button"
            >
              ›
            </button>
            <button
              aria-label=""
              className="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
              onClick={[Function]}
              type="button"
            >
              »
            </button>
          </div>
          <div
            className="react-calendar__month-view "
          >
            <div
              style={
                Object {
                  "alignItems": "flex-end",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "flexGrow": 1,
                    "width": "100%",
                  }
                }
              >
                <div
                  className="react-calendar__month-view__weekdays"
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": undefined,
                      "flexWrap": "no-wrap",
                    }
                  }
                >
                  <div
                    className="react-calendar__month-view__weekdays__weekday"
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                  >
                    <abbr
                      aria-label="Monday"
                      title="Monday"
                    >
                      Mon
                    </abbr>
                  </div>
                  <div
                    className="react-calendar__month-view__weekdays__weekday"
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                  >
                    <abbr
                      aria-label="Tuesday"
                      title="Tuesday"
                    >
                      Tue
                    </abbr>
                  </div>
                  <div
                    className="react-calendar__month-view__weekdays__weekday"
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                  >
                    <abbr
                      aria-label="Wednesday"
                      title="Wednesday"
                    >
                      Wed
                    </abbr>
                  </div>
                  <div
                    className="react-calendar__month-view__weekdays__weekday"
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                  >
                    <abbr
                      aria-label="Thursday"
                      title="Thursday"
                    >
                      Thu
                    </abbr>
                  </div>
                  <div
                    className="react-calendar__month-view__weekdays__weekday"
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                  >
                    <abbr
                      aria-label="Friday"
                      title="Friday"
                    >
                      Fri
                    </abbr>
                  </div>
                  <div
                    className="react-calendar__month-view__weekdays__weekday"
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                  >
                    <abbr
                      aria-label="Saturday"
                      title="Saturday"
                    >
                      Sat
                    </abbr>
                  </div>
                  <div
                    className="react-calendar__month-view__weekdays__weekday"
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                  >
                    <abbr
                      aria-label="Sunday"
                      title="Sunday"
                    >
                      Sun
                    </abbr>
                  </div>
                </div>
                <div
                  className="react-calendar__month-view__days"
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": undefined,
                      "flexWrap": "wrap",
                    }
                  }
                >
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="September 30, 2019"
                    >
                      30
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 1, 2019"
                    >
                      1
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 2, 2019"
                    >
                      2
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 3, 2019"
                    >
                      3
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 4, 2019"
                    >
                      4
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 5, 2019"
                    >
                      5
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 6, 2019"
                    >
                      6
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 7, 2019"
                    >
                      7
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 8, 2019"
                    >
                      8
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 9, 2019"
                    >
                      9
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 10, 2019"
                    >
                      10
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 11, 2019"
                    >
                      11
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 12, 2019"
                    >
                      12
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 13, 2019"
                    >
                      13
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 14, 2019"
                    >
                      14
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 15, 2019"
                    >
                      15
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 16, 2019"
                    >
                      16
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 17, 2019"
                    >
                      17
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 18, 2019"
                    >
                      18
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 19, 2019"
                    >
                      19
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 20, 2019"
                    >
                      20
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 21, 2019"
                    >
                      21
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 22, 2019"
                    >
                      22
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 23, 2019"
                    >
                      23
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 24, 2019"
                    >
                      24
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 25, 2019"
                    >
                      25
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 26, 2019"
                    >
                      26
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 27, 2019"
                    >
                      27
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 28, 2019"
                    >
                      28
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 29, 2019"
                    >
                      29
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 30, 2019"
                    >
                      30
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="October 31, 2019"
                    >
                      31
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="November 1, 2019"
                    >
                      1
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="November 2, 2019"
                    >
                      2
                    </abbr>
                  </button>
                  <button
                    className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                    onClick={[Function]}
                    onFocus={null}
                    onMouseOver={null}
                    style={
                      Object {
                        "flexBasis": "14.285714285714286%",
                        "marginLeft": null,
                        "maxWidth": "14.285714285714286%",
                        "overflow": "hidden",
                      }
                    }
                    type="button"
                  >
                    <abbr
                      aria-label="November 3, 2019"
                    >
                      3
                    </abbr>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <label>
        Anything you care to add?
         
        <span
          style={
            Object {
              "color": "smokewhite",
              "fontStyle": "italic",
            }
          }
        >
          (optional)
        </span>
      </label>
      <input
        name="content"
        onChange={[Function]}
        type="text"
        value=""
      />
      <div
        id="required-fields"
      >
        <p
          style={
            Object {
              "color": "smokewhite",
              "fontStyle": "italic",
            }
          }
        >
          (* indicates required field)
        </p>
      </div>
      <label>
        Rate your experience of the show*
      </label>
      <div
        className="ratings"
      >
        <input
          className={null}
          id="rating"
          max="5"
          min="1"
          name="rating"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
          value=""
        />
      </div>
      <button
        disabled={true}
        id="submit-review-button"
        type="disabled"
      >
        Submit
      </button>
    </form>
    <div
      id="error-message"
    />
  </div>,
  <div
    className="add-reviews-push"
  />,
]
`;
